client/src/components/Home.js

```diff
@@ -65,60 +64,93 @@ const Home = ({ user, logout }) => {
   const postMessage = (body) => {
     try {
       const data = saveMessage(body);
-
-      if (!body.conversationId) {
-        addNewConvo(body.recipientId, data.message);
-      } else {
-        addMessageToConversation(data);
-      }
-
-      sendMessage(data, body);
+      data.then(res => {
+        if (!body.conversationId) {
+          addNewConvo(body.recipientId, res.message);
+        } else {
+          addMessageToConversation(res, body.recipientId);
+        }
+        sendMessage(res, body);
+      });
     } catch (error) {
       console.error(error);
     }
   };
 
+
   const addNewConvo = useCallback(
     (recipientId, message) => {
-      conversations.forEach((convo) => {
+      let newState = [...conversations];
+      let currentConv = null
+      conversations.forEach((convo, index) => {
         if (convo.otherUser.id === recipientId) {
-          convo.messages.push(message);
+          convo.messages.unshift(message);
           convo.latestMessageText = message.text;
-          convo.id = message.conversationId;
+          currentConv = convo;
+          const idRandom = (Math.random() + 1).toString(36).substring(7);
+          currentConv.id = idRandom;
+          newState.splice(index, 1);
+          return;
         }
       });
-      setConversations(conversations);
+      console.log(currentConv);
+      if (currentConv != null) newState.unshift(currentConv);
```

Comment:
The most recent conversation already is moved to the top of the conversations by default, we don't need this to happen in real-time 🙂